subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DependencyManagementBuildScopeServices.java

```diff
@@ -188,6 +192,18 @@ ComponentIdentifierFactory createComponentIdentifierFactory(BuildState currentBu
         return new DefaultComponentIdentifierFactory(buildRegistry.getBuild(currentBuild.getBuildIdentifier()));
     }
 
+    VersionComparator createVersionComparator(FeaturePreviews featurePreviews, ListenerManager listenerManager) {
```

Comment:
:o: : I think this could be build session scoped

Reply:
Distinction not available for dependency management services for now, not changing.